rust_src/src/keyboard.rs

```diff
@@ -220,4 +229,52 @@ pub extern "C" fn rust_syms_of_keyboard() {
     defvar_kboard!(Vlast_repeatable_command_, "last-repeatable-command");
 }
 
+/// Produce default output for unhandled error message.
+/// Default value of `command-error-function'.
+#[lisp_fn]
+pub fn command_error_default_function(
+    data: LispObject,
+    context: LispStringRef,
+    signal: LispObject,
+) -> LispObject {
+    let sf = selected_frame();
+
+    // If the window system or terminal frame hasn't been initialized
+    // yet, or we're not interactive, write the message to stderr and
+    // exit.
+    if sf.glyphs_initialized_p()
+        // The initial frame is a special non-displaying frame. It
+        // will be current in daemon mode when there are no frames to
+        // display, and in non-daemon mode before the real frame has
+        // finished initializing.  If an error is thrown in the latter
+        // case while creating the frame, then the frame will never be
+        // displayed, so the safest thing to do is write to stderr and
+        // quit.  In daemon mode, there are many other potential
+        // errors that do not prevent frames from being created, so
+        // continuing as normal is better in that case.
+        || (!is_daemon() && sf.output_method() == output_method::output_initial)
+        || unsafe {globals.noninteractive1 }
+    {
+        unsafe {
+            print_error_message(
+                data,
+                Qexternal_debugging_output,
+                context.const_sdata_ptr(),
+                signal,
+            );
+            Fterpri(Qexternal_debugging_output, Qnil);
+            Fkill_emacs(LispObject::from(-1));
+        }
+    } else {
+        unsafe {
+            clear_message(true, false);
+            Fdiscard_input();
+            message_log_maybe_newline();
+            ding_internal(true);
+            print_error_message(data, Qt, context.const_sdata_ptr(), signal);
+        }
+    }
+    Qnil
```

Comment:
There is no need for the plain `Qnil` return. Leave off the return type in the function definition and the right thing will happen.